File: libraryCoreJava/src/main/java/com/juqitech/moretickets/core/base/impl/BaseModel.java
package com.juqitech.moretickets.core.base.impl;

import androidx.annotation.NonNull;

import com.juqitech.moretickets.core.base.IBaseModel;
import com.juqitech.moretickets.core.http.HttpClient;
import com.juqitech.moretickets.core.util.LogUtil;

/**
 * @author zhiqiang
 * @date 2019-06-04
 * @desc
 */
public class BaseModel<T> implements IBaseModel {

    /**
     * apiService
     * 用于 retrofit 请求网络
     */
    protected T mApiService;

    public BaseModel(@NonNull Class<T> clazz) {
        this.mApiService = HttpClient.create(clazz);
    }

    @Override
    public void onDestroy() {
        LogUtil.i("BaseModel", "onDestroy");
        mApiService = null;
    }

}

File: libraryCore/src/main/java/com/juqitech/moretickets/core/base/impl/BaseModel.kt
package com.juqitech.moretickets.core.base.impl

import com.juqitech.moretickets.core.base.IModel
import com.juqitech.moretickets.core.http.HttpClient
import com.juqitech.moretickets.core.util.LogUtil

/**
 * @author zhanfeng
 * @date 2019-06-04
 * @desc
 */
open class BaseModel<T>(clazz: Class<T>) : IModel {

    /**
     * apiService
     * 用于 retrofit 请求网络
     */
    protected var mApiService: T? = null

    init {
        this.mApiService = HttpClient.create(clazz)
    }

    override fun onDestroy() {
        LogUtil.i("BaseModel", "onDestroy")
        mApiService = null
    }

}
